sdk/docs/storage/README.md

```diff
@@ -5,49 +5,67 @@ Azure Storage Blobs library (`az_storage_blobs`) provides abstractions, and help
 Use the Azure SDK Storage Blobs Library for Embedded C to work with Azure Storage Blobs.
 
 * Upload blobs
-* Create and modify blob metadata
 
 ## Getting Started
 
-### Install the Package
+The Azure Storage Blobs Client library is created to facilitate connectivity to Azure Storage Blobs services using the HTTP stack of the user's choice. This means that this SDK is **NOT** a platform but instead is a true SDK library.
 
-TODO link to the vcpkg
+### Docs
 
-### Authenticate the Client
+For API documentation, please see the doxygen generated docs [here][azure_sdk_for_c_doxygen_docs]. You can find the Storage Blobs specific docs by navigating to the **Files -> File List** section near the top and choosing any of the header files prefixed with `az_storage_blobs`.
 
-TODO
+### Build
 
-### Get Credentials
+The Azure Storage Blobs library is compiled following the same steps listed on the root [README][azure_sdk_for_c_readme] documentation, under ["Getting Started Using the SDK"][azure_sdk_for_c_readme_getting_started].
 
-TODO
+The library targets made available via CMake are the following:
 
-### Create Client
+- `az::storage::blobs` - For Azure Storage Blobs features ([API documentation here][azure_sdk_for_c_doxygen_docs])
 
-TODO
+### Samples
+
+This [page][samples_storage_blobs_readme] explains samples for the Azure Embedded C SDK Storage Blobs Client.
+
+
+### Prerequisites
+
+For compiling the Azure SDK for Embedded C for the most common platforms (Windows and Linux), no further prerequisites are necessary.
+Please follow the instructions in the [Getting Started](#Getting-Started) section above.
+For compiling for specific target devices, please refer to their specific toolchain documentation.
 
 ## Key Concepts
 
-TODO
+### Authentication
+
+The embedded C SDK currently supports SAS based authentication.  See [this page][storage_access_control_sas] for information on creating SAS tokens.
+The client credential should be set to `AZ_CREDENTIAL_ANONYMOUS` when using SAS tokens.
 
-## Examples
+### Creating the Storage Client
 
-TODO
+To use the storage client the first action is to initialize the client with `az_storage_blobs_blob_client_init`.
+```C
+  az_storage_blobs_blob_client client;
+  az_storage_blobs_blob_client_options options = az_storage_blobs_blob_client_options_default();
 
-## Troubleshooting
+  az_storage_blobs_blob_client_init(
+          &client, az_span_from_str(getenv(URI_ENV)), AZ_CREDENTIAL_ANONYMOUS, &options);
```

Comment:
We should just pass in `NULL` for options, to keep the getting started simple and "one-liner".

Reply:
I prefer using the explicit value.  It makes it explicit what we are doing.

Reply:
I think not allowing `NULL`, as in keeping the options, optional, is a usability concern.
We have the pattern of passing in `NULL` to mean `default` in a whole bunch of existing SDK APIs. I think we should make sure both work, but that is related to the API and not docs (unrelated to this PR).

But you are right, leaving the option in the code snippet in the docs and being explicit is reasonable.

Remove the `NULL` parameter below in `az_storage_blobs_blob_upload`?

Reply:
We can't remove that NULL parameter.  That is the options for upload which are optional

Reply:
By remove, I mean state that parameter explicitly. The same rationale you are applying here.